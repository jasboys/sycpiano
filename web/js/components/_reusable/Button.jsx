import 'less/_reusable/button.less';

import React from 'react';
import classNames from 'classnames';

const Button = ({ children, extraClasses, onClick }) => (
    <div
        className={classNames('Button', { [extraClasses]: !!extraClasses })}
        onClick={onClick}
    >
        {children}
    </div>
);

export default Button;
