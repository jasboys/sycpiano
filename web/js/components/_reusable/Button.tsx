import 'less/_reusable/button.less';

import * as React from 'react';

interface ButtonProps {
    extraClasses: string[];
    onClick: (event: React.MouseEvent<HTMLElement>) => void;
}

function mergeClasses(classes: string[]): string {
    return classes.reduce((accumulator: string, value: string): string => {
        return accumulator += `${value} `;
    }, '');
}

const Button: React.SFC<ButtonProps & React.HTMLAttributes<HTMLElement>> = ({ children, extraClasses, onClick }) => (
    <div
        className={mergeClasses(['Button', ...extraClasses])}
        onClick={onClick}
    >
        {children}
    </div>
);

export default Button;
