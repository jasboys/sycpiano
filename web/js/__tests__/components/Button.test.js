import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Button from '@/js/components/_reusable/Button.jsx';

it('renders correctly', () => {
    const tree = renderer.create(
        <Button />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders with extra class', () => {
    const tree = renderer.create(
        <Button
            extraClasses='extra-class'
            onClick={() => {}}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders with onClick', () => {
    const mockOnClick = jest.fn();
    const button = shallow(<Button onClick={mockOnClick} />);
    button.find('.Button').simulate('click');
    expect(mockOnClick.mock.calls.length).toBe(1);
});
