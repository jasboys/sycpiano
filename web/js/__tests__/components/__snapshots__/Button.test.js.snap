// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Button"
  onClick={undefined}
/>
`;

exports[`renders with extra class 1`] = `
<div
  className="Button extra-class"
  onClick={[Function]}
/>
`;
